Texture.py

```diff
@@ -226,14 +255,101 @@ def read_texture_parameters(xbox: Xbox) -> TextureParameters:
         format_color=fmt_color,
         depth_pitch=depth_pitch,
         depth_offset=depth_offset,
-        format_depth=None,
+        format_depth=fmt_depth,
         surface_type=surface_type,
         swizzle_unk=swizzle_unk,
         swizzle_unk2=swizzle_unk2,
         swizzled=swizzled,
     )
 
 
+def dump_zeta(data, offset, pitch, fmt_depth, width, height):
+    """Convert the zeta buffer at the given offset into a PIL.Image."""
+    depth_img = Image.new("RGB", (width, height))
+    stencil_img = None
+
+    if fmt_depth == 0x2D:
+        info = (True, True, Z16)
+    elif fmt_depth == 0x31:
+        info = (False, True, Z16)
+    elif fmt_depth == 0x2C:
+        info = (True, False, Z16)
+    elif fmt_depth == 0x30:
+        info = (False, False, Z16)
+    elif fmt_depth == 0x2B:
+        info = (True, True, Z24S8)
+        stencil_img = Image.new("RGB", (width, height))
+    elif fmt_depth == 0x2F:
+        info = (False, True, Z24S8)
+        stencil_img = Image.new("RGB", (width, height))
+    elif fmt_depth == 0x2A:
+        info = (True, False, Z24S8)
+        stencil_img = Image.new("RGB", (width, height))
+    elif fmt_depth == 0x2E:
+        info = (False, False, Z24S8)
+        stencil_img = Image.new("RGB", (width, height))
+    else:
+        raise Exception("Unknown depth format :0x%X" % fmt_depth)
+
+    swizzle = info[0]
+    is_float = info[1]
+    fmt = info[2]
+
+    # FIXME: Avoid this nasty ~~convience feature~~ hack
+    if pitch == 0:
+        pitch = width * fmt.bpp // 8
+
+    assert len(data) == pitch * height
+
+    # FIXME: Does this work?
+    if False and swizzle:
+        data = xbox.Unswizzle(data, fmt.bpp, (width, height), pitch)
+
+    depth_values = {}
+    depth_max = 0
+    depth_min = 0xFFFFFF
+    stencil_values = {}
+
+    for y in range(height):
+        for x in range(width):
+            pixel_offset = y * pitch + x * fmt.bpp // 8
+            z_data = data[pixel_offset : pixel_offset + fmt.bpp // 8]
+            zeta = int.from_bytes(z_data, "little")
+            depth = (zeta & fmt.depth_mask) >> fmt.depth_shift
+            stencil = zeta & fmt.stencil_mask
+
+            if is_float:
+                depth = fmt.convert_float(depth)
+
+            depth_values[x, y] = depth
+            stencil_values[x, y] = 255 if stencil != 0 else 0
+
+            if depth > depth_max:
+                depth_max = depth
+
+            if depth < depth_min:
+                depth_min = depth
+
+    # Map depth values to [0, 255]
+    assert depth_max >= depth_min
+    depth_max -= depth_min
+    depth_scaler = 255 / depth_max if depth_max > 0 else 255
+    depth_pixels = depth_img.load()
+    if stencil_img is not None:
```

Comment:
Nit: I don't think you need to explicitly check against `None` here, just `if stencil_img` would be more pythonic since anything falsey is not going to be callable.